hbase_sql/hbase1.2.0/src/main/java/com/splicemachine/compactions/SpliceDefaultCompactor.java

```diff
@@ -506,4 +500,209 @@ protected InternalScanner createScanner(Store store, List<StoreFileScanner> scan
+                }
+            } else {
+                // Family does not provide key material, create a random key
+                cipher = Encryption.getCipher(conf, cipherName);
+                if (cipher == null) {
+                    throw new RuntimeException("Cipher '" + cipherName + "' is not available");
+                }
+                key = cipher.getRandomKey();
+            }
+            Encryption.Context cryptoContext = Encryption.newContext(conf);
+            cryptoContext.setCipher(cipher);
+            cryptoContext.setKey(key);
+            return cryptoContext;
+        } else
+            return Encryption.Context.NONE;
+    }
+
+    /**
+     * Creates a writer for a new file in a temporary directory.  This is pulled forward from DefaultCompactor
+     * to handle some scoping issues.
+     *
+     * @param fd The file details.
+     * @return Writer for a new StoreFile in the tmp dir.
+     * @throws IOException
+     */
+    protected StoreFile.Writer createTmpWriter(FileDetails fd, boolean shouldDropBehind, InetSocketAddress[] favoredNodes)
+            throws IOException {
+
+        // When all MVCC readpoints are 0, don't write them.
+        // See HBASE-8166, HBASE-12600, and HBASE-13389.
+
+        return createWriterInTmp(fd.maxKeyCount, this.compactionCompression,
+            /* isCompaction = */ true,
+            /* includeMVCCReadpoint = */ fd.maxMVCCReadpoint > 0,
+            /* includesTags = */ fd.maxTagsLength > 0,
+            /* shouldDropBehind = */ shouldDropBehind,
+                favoredNodes);
+    }
+
+    /**
+     *
+     * This only overwrites favored nodes when there are none supplied.  I believe in later versions the favoredNodes are
+     * populated for region groups.  When this happens, we will pass those favored nodes along.  Until then, we attempt to put the local
+     * node in the favored nodes since sometimes Spark Tasks will run compactions remotely.
+     *
+     * @return
+     * @throws IOException
+     */
+    protected InetSocketAddress[] getFavoredNodes() throws IOException {
+        try {
+            RegionServerServices rsServices = (RegionServerServices)FieldUtils.readField(((HStore) store).getHRegion(), "rsServices", true);
+            InetSocketAddress[] returnAddresses = (InetSocketAddress[]) MethodUtils.invokeMethod(rsServices,"getFavoredNodesForRegion",store.getRegionInfo().getEncodedName());
+            if ( (returnAddresses == null || returnAddresses.length == 0)
+                    && store.getFileSystem() instanceof HFileSystem
+                    && ((HFileSystem)store.getFileSystem()).getBackingFs() instanceof DistributedFileSystem) {
+                String[] txvr = conf.get("dfs.datanode.address").split(":"); // hack
+                if (txvr.length == 2) {
+                    returnAddresses = new InetSocketAddress[1];
+                    returnAddresses[0] = new InetSocketAddress(hostName, Integer.parseInt(txvr[1]));
```

Comment:
If dfs.datanode.address is malformed, like hostname:abc, we'll throw an exception instead of allowing the compaction to complete with a warning.  Is this intended?